tests/functional/email_opt_in.js

```diff
@@ -63,170 +58,84 @@ define([
+        .then(openVerificationLinkInSameTab(email, 0))
 
+        .then(testElementExists('#communication-preferences.basket-ready'))
         .then(waitForBasket(email))
-
-        .findByCssSelector('#communication-preferences .settings-unit-toggle')
-          .click()
-        .end()
-
-        .then(FunctionalHelpers.visibleByQSA('#communication-preferences .settings-unit-details'))
-
-        .then(testOptedIn(self))
+        .then(click('#communication-preferences .settings-unit-toggle'))
+        .then(visibleByQSA('#communication-preferences .settings-unit-details'))
+        .then(testOptedIn())
 
         // user signed up to basket, so has a preferences URL
-        .findByCssSelector('#preferences-url')
-        .end()
+        .then(testElementExists('#preferences-url'))
 
         // ensure the changes stick across refreshes
         .refresh()
 
-        .findByCssSelector('#communication-preferences.basket-ready')
-        .end()
-
-        .then(FunctionalHelpers.visibleByQSA('#communication-preferences .settings-unit-details'))
-
-        .then(testOptedIn(self))
-
-        .findByCssSelector('#marketing-email-optin')
-          .click()
-        .end()
-
-        .then(FunctionalHelpers.testSuccessWasShown(self))
+        .then(testElementExists('#communication-preferences.basket-ready'))
+        .then(visibleByQSA('#communication-preferences .settings-unit-details'))
+        .then(testOptedIn())
+        .then(click('#marketing-email-optin'))
+        .then(testSuccessWasShown(this))
 
         // ensure the opt-out sticks across refreshes
         .refresh()
 
-        .findByCssSelector('#communication-preferences.basket-ready')
-        .end()
-
-        .findByCssSelector('#communication-preferences .settings-unit-toggle')
-          .click()
-        .end()
+        .then(testElementExists('#communication-preferences.basket-ready'))
+        .then(click('#communication-preferences .settings-unit-toggle'))
+        .then(visibleByQSA('#communication-preferences .settings-unit-details'))
 
-        .then(FunctionalHelpers.visibleByQSA('#communication-preferences .settings-unit-details'))
-
-        .then(testNotOptedIn(self));
+        .then(testNotOptedIn());
     },
 
-    'do not opt in to the marketing email': function () {
-      var self = this;
-      return FunctionalHelpers.openPage(this, PAGE_URL, '#fxa-signup-header')
-        .then(function () {
-          return fillOutSignUp(self, email, PASSWORD, { optInToMarketingEmail: false });
-        })
-
-        .findByCssSelector('#fxa-confirm-header')
-        .end()
-
-        .then(function () {
-          return FunctionalHelpers.getVerificationLink(email, 0);
-        })
-        .then(function (verificationLink) {
-          return self.remote.get(require.toUrl(verificationLink));
-        })
-
-        .findByCssSelector('#communication-preferences.basket-ready')
-        .end()
+    'opt in to marketing email from settings': function () {
```

Comment:
The change in test title threw me off a little. Would this be better off as two separate tests, `do not opt in to the marketing email` and `opt in to marketing email from settings`. Or maybe rename this to `opt in to marketing email from settings not from signup`? I could be overthinking :)